### PR TITLE
Product-service

### PR DESCRIPTION
**Pull request by DrDiman**
Everything was done:

- poiproduct-service serverless config contains configuration for 2 lambda functions,
- The getProductsById AND getProductsList lambda functions return a correct response code (POINT2)
-  Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend.

Additional (optional) tasks were done:
+1 - Async/await is used in lambda functions
+1 - ES6 modules are used for product-service implementation
+1 - Webpack is configured for product-service
+1 - SWAGGER documentation is created for product-service
+1 - Lambda handlers are covered by basic UNIT tests (NO infrastructure logic is needed to be covered) (used JEST)
+1 - Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase.
+1 - Main error scenarious are handled by API ("Product not found" error, try catch blocks are used in lambda handlers).

Link to product-service API - https://a7o6p63xnd.execute-api.eu-west-1.amazonaws.com/dev/products

LInk to FE MR - https://github.com/DrDiman/shop-react-redux-cloudfront/pull/2

SWAGGER file is provided and committed as openapi.yml file

### CODE REVIEW

**Comment by ReshetovItsMe** [comment]
How can I see swagger ui?
I also researched your variant of implementation (by serverless-openapi-documentation), but I thought it isn't right way because we can't get access to ui.
What's pros from that configuration? Why we can't use usual swagger config.

**Comment by DrDiman** [comment]
> How can I see swagger ui?
> I also researched your variant of implementation (by serverless-openapi-documentation), but I thought it isn't right way because we can't get access to ui.
> What's pros from that configuration? Why we can't use usual swagger config.

@ReshetovItsMe 
I've never created swagger doc before, so I can't list pros and cons. I agree with u that it's not so useful and user friendly - I used this plugin as proposed by lectures. As I clarified it's enough to have yml file for now in this task. 

could u provide links to the right (better) way for creation swagger doc?

**Comment by ReshetovItsMe** [comment]
> > How can I see swagger ui?
> > I also researched your variant of implementation (by serverless-openapi-documentation), but I thought it isn't right way because we can't get access to ui.
> > What's pros from that configuration? Why we can't use usual swagger config.
> 
> @ReshetovItsMe
> I've never created swagger doc before, so I can't list pros and cons. I agree with u that it's not so useful and user friendly - I used this plugin as proposed by lectures. As I clarified it's enough to have yml file for now in this task.
> 
> could u provide links to the right (better) way for creation swagger doc?

@DrDiman, you can just write your own swagger.yml in [swagger editor](https://editor.swagger.io/?_ga=2.75438101.625418254.1626244427-2142695032.1626244427). And that's why I don't understand the reasons to use this plugin for serverless. 
I thought that it will automatically generate swagger-ui, but it's not. So, that's why I didn't finished this task.
Maybe we should ask this question in channel.

**Comment by ReshetovItsMe** [comment]
also I can't get any response from https://a7o6p63xnd.execute-api.eu-west-1.amazonaws.com/dev/products

File: product-service/src/functions/getProductById/getProductById.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+const handler = async (event) => {
+  try {
+    const { id } = event.pathParameters
+    const mockDbResponse = await getMockProductList()
+    const product = mockDbResponse.find((p) => p.id === id)
+    return product
+      ? formatJSONResponse(product, 200)
+      : formatJSONResponse(null, 404)
+  } catch (error) {
+    return formatJSONResponse(null, 500)
```
Comment (ReshetovItsMe) [inline product-service/src/functions/getProductById/getProductById.js:14]: I think here and above we should return correct messages like "Can't access to database" and "Can't find product with id ...."
Same for producttList

File: product-service/src/schemas/productsList.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  "definitions": {},
+  "title": "Products List",
+  "type": "array",
+  "default": [],
+  "items": {
+    "title": "Items",
+    "type": "object",
+    "required": ["count", "description", "id", "price", "title"],
+    "properties": {
+      "count": {
```
Comment (MikeShakmakov) [inline product-service/src/schemas/productsList.json:11]: looks like copy paste, I think its better to reuse product model here

File: product-service/openapi.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        title: Items
+        type: object
+        required:
+          - count
+          - description
+          - id
+          - price
+          - title
+        properties:
+          count:
```
Comment (MikeShakmakov) [inline product-service/openapi.yml:18]: and copy paste is here too

**Pull request merged**